Build/Build.cs

```diff
@@ -258,11 +272,36 @@ class Build : NukeBuild
         });
 
     Target SpellCheck => _ => _
+        .OnlyWhenDynamic(() => DocumentationChangesDetected())
         .Executes(() =>
         {
             Node($"{YarnCli} install --silent", workingDirectory: RootDirectory);
             Node($"{YarnCli} --silent run cspell --no-summary", workingDirectory: RootDirectory,
                 customLogger: (_, msg) => Error(msg));
         });
+
+    bool DocumentationChangesDetected()
```

Comment:
⛏️ (nitpick) I prefer `HasDocumentationChanges` and `HasSourceChanges`
🔧 Make it properties